build/scripts/ensure-toolset.ps1

```diff
@@ -0,0 +1,58 @@
+param ()
+set-strictmode -version 2.0
+$ErrorActionPreference="Stop"
+
+$nugetVersion = "3.6.0-beta1"
+$msbuildToolsetVersion = "0.1.3-alpha"
+$repoDir = [IO.Path]::GetFullPath((join-path $PSScriptroot "..\.."))
+$toolsDir = join-path $repoDir "Binaries\Toolset"
+
+# Download the specified tool.  The download name needs to be a version name that
+# will be different when we change the version strings above.  That way the script
+# does work when the strings change vs. using the already downloaded and cached
+# values.
+function download-tool() {
+    param ([string]$url, [string]$fileName, [string]$downloadName)
+
+    $scratchDir = join-path $toolsDir "Scratch"
+    mkdir $scratchDir -ErrorAction SilentlyContinue | out-null
+    
+    $filePath = join-path $toolsDir $fileName
+    $downloadPath = join-path $scratchDir $downloadName
+
+    if ((test-path $filePath) -and (test-path $downloadPath)) {
+        return $false
+    }
+        
+    write-host "Downloading $fileName"
+    $webClient = New-Object -TypeName "System.Net.WebClient"
+    $webClient.DownloadFile($url, $downloadPath)
+    cp $downloadPath $filePath
+    return $true
+}
+
+try {
+    mkdir $toolsDir -ErrorAction silent | out-null
+    pushd $toolsDir
+    
+    $nugetUrl = "https://dist.nuget.org/win-x86-commandline/v$nugetVersion/NuGet.exe"
+    if (download-tool $nugetUrl "NuGet.exe" "NuGet-$nugetVersion.exe") {
+        cp "NuGet.exe" $repoDir
+    }
+
+    $msbuildUrl = "https://jdashstorage.blob.core.windows.net/msbuild/msbuild-$msbuildToolsetVersion.zip" 
```

Comment:
> Because it's the future :)

It's not the future though.  The current design will **never** work for our desktop build.  We are forever working with desktop MSBuild in our repo.  

> I assume we want to run tests on CoreCLR on Windows. The easiest way to do that is to run on dotnet cli. 

Sure.  It will be *possibly* be an advantage on the platforms where we run CoreClr tests.  Today we have pretty much 0 investment in that though.  

> The less differences between builds for different platforms the better.

Doesn't seem like a good argument here.  We cannot eliminate desktop MSBuild ever.  Hence there will always be differences.  Using CLI to restore then dekstop to build seems like an unnecessary complication.  

> I don't see why microbuild build task couldn't be a portable library.

I doubt we can convince Microbuild, VS SDK, WPF, Debugger, etc ... to all have fully portable tools and MSBuild plugins.  They have 0 incentive to make this transition other than "to make Roslyn happy".  There is no other upside for them. 

I agree that we can eventually push Microbuild on this.  They do have some incentive.  The other teams though have no incentive and I doubt they will move.

> Are there any more msbuild tasks that we need? VSIX tooling runs as a separate .exe, no?